chain/ethereum/src/trigger.rs

```diff
@@ -294,6 +295,9 @@ impl EthereumTrigger {
 
             // Unfiltered block triggers match any data source address.
             EthereumTrigger::Block(_, EthereumBlockTriggerType::Every) => None,
+            EthereumTrigger::Block(_, EthereumBlockTriggerType::Polling(address, _)) => {
+                Some(address)
+            }
```

Comment:
In principle, I think `EthereumBlockTriggerType` should not carry the address, and `None` should be returned here. As a block trigger, in principle, should not be associated to any specific address since it represents the whole block. `EthereumBlockTriggerType::WithCallTo` is an exception to this principle, and I'd prefer not to add another one.